lxd/db/cluster/entities.go

```diff
@@ -384,47 +388,63 @@ func GetEntityURLs(ctx context.Context, tx *sql.Tx, projectName string, entityTy
 	// If the project is not empty, each statement will need an argument for the project name.
 	// Additionally, pre-populate the result map as we know the entity types in advance (this is so that we don't have
 	// to check and assign on each loop iteration when scanning rows).
-	if len(entityTypes) == 0 && projectName == "" {
-		for entityType, stmt := range entityStatementsAll {
-			stmts = append(stmts, stmt)
-			result[entityType] = make(map[int]*api.URL)
+	if len(entityTypesFilter) == 0 && projectName == "" {
+		for _, dbEntityType := range entityTypes {
+			stmt := dbEntityType.AllURLs()
+			if stmt != "" {
```

Comment:
Hmmm... for this one perhaps but not for `URLsByProject` for example, as not all entities are project specific.

The implementation of the "server" `entityType` in the cluster package returns an empty string for `AllURLs` because there is no database query should be run for it (the caller should know that it's `/1.0`).